platform/starbase/src/jsonrpc/methods/getAuthorizedAccounts.ts

```
@@ -0,0 +1,68 @@
+import createEdgesClient from '@kubelt/platform-clients/edges'
+import { AccessRComp } from '@kubelt/urns/access'
+import { Context } from '../context'
+import { EDGE_AUTHORIZES } from '@kubelt/platform.access/src/constants'
+// import { Graph } from '@kubelt/types'
+// import { inputValidators } from '@kubelt/platform-middleware'
+import { z } from 'zod'
+
+// Input
+// -----------------------------------------------------------------------------
+
+export const GetAuthorizedAccountsMethodInput = z.object({
+  client: z.string(),
+})
+
+export type GetAuthorizedAccountsParams = z.infer<
+  typeof GetAuthorizedAccountsMethodInput
+>
+
+// Output
+// -----------------------------------------------------------------------------
+
+export const GetAuthorizedAccountsMethodOutput = z.array(
+  z.object({
+    accountURN: z.string(),
+    timestamp: z.number(),
+  })
+)
+
+// Method
+// -----------------------------------------------------------------------------
+
+export const getAuthorizedAccounts = async ({
+  input,
+  ctx,
+}: {
+  input: GetAuthorizedAccountsParams
+  ctx: Context
+}) => {
+  const edgesClient = createEdgesClient(ctx.Edges)
+
+  const edgesResult = await edgesClient.getEdges.query({
+    query: {
+      tag: EDGE_AUTHORIZES,
+    },
+    // set limit to not query the whole db
+    opt: { limit: 8 },
+  })
+
+  const mappedEdges = edgesResult.edges
+    .filter(
+      (edge) =>
+        (edge.dst.rc as AccessRComp).client_id &&
+        (edge.dst.rc as AccessRComp).client_id === input.client
```

It'd be great if `getEdges` can take type parameters which would eliminate (repetitive) type assertions. Another alternative would be a conditional expression (or an if statement) to fail early or ensure type safety on the property.

---

tbh I think now it's easier to keep it this way